scripts/launch_cluster.py

```diff
@@ -0,0 +1,640 @@
+
+            # check vpc create complete
+            describe_vpcs_request = vpc_20160428_models.DescribeVpcsRequest(
+                vpc_id=vpc_id,
+                region_id=self._region
+            )
+            while True:
+                res = self._vpc.describe_vpcs(describe_vpcs_request)
+                if res.body.vpcs.vpc[0].status == "Available":
+                    break
+
+            # get zones of region
+            click.echo("Creating switch...")
+            describe_zones_request = vpc_20160428_models.DescribeZonesRequest(region_id=self._region)
+            desc_zone_res = self._vpc.describe_zones(describe_zones_request)
+            zone_id = desc_zone_res.body.zones.zone[0].zone_id
+            # create switch
+            create_vswitch_request = vpc_20160428_models.CreateVSwitchRequest(
+                zone_id=zone_id,
+                cidr_block=vswitch_cidr_block,
+                vpc_id=vpc_id,
+            )
+            vswitch_res = self._vpc.create_vswitch(create_vswitch_request)
+            vswitch_ids = [vswitch_res.body.v_switch_id]
+            
+            # check vswitch create complete
+            describe_vswitch_request = vpc_20160428_models.DescribeVSwitchAttributesRequest(
+                v_switch_id=vswitch_ids[0]
+            )
+            while True:
+                res = self._vpc.describe_vswitch_attributes(describe_vswitch_request)
+                if res.body.status == "Available":
+                    break
+
+        click.echo("Get vpc ID: " + vpc_id)
+        click.echo("Get vswitch ids: " + str(vswitch_ids))
+        return vpc_id, vswitch_ids
+    
+    def wait_cluster_ready(self, cluster_id):
+        # Going to give up after 40 times 20 seconds.  
+        cnt=40
+        while True:
+            # Wait 20 seconds
+            time.sleep(20)
+            # Get cluster status
+            response = self._eks.describe_cluster_detail(cluster_id)
+            status = response.body.state
+            click.echo("Checking cluster status: %s" % status)
+            # Is it active? If so break out of loop
+            if status == "running":
+                break
+            elif status == "failed":
+                click.echo("Cluster create failed.")
+                sys.exit(1)
+            # Maybe give up after so many goes.
+            cnt = cnt - 1
+            if cnt <= 0:
+                click.echo("Given up waiting for cluster to go RINNING.")
+                sys.exit(1)
+
+
+@click.command()
+@click.option("--type", type=click.Choice(["aws", "aliyun"], case_sensitive=False), 
+              help="Cloud type to launch cluster.")
+@click.option("--id", help="The access_key_id of cloud.")
+@click.option("--secret", help="The access_key_secret of cloud.")
+@click.option("--region", help="The region id.")
+@click.option("--output", help="The kube config file output path.", default=os.environ["HOME"] + "/.kube/config",
```

Comment:
If user's home already contains a config, would you overwrite that?
If so, please make a backup and tell user where it is.

Reply:
done.